### PR TITLE
feat: APPS-2583-2582 Fix block StaffSubjectLibrarian & Convert SectionStaffSubjectLibrarian to vue3

### PR DESCRIPTION
**Pull request by jendiamond**
Connected to [2583](https://uclalibrary.atlassian.net/jira/software/c/projects/APPS/boards/12?assignee=712020%3Af4539cc0-90d3-4445-9de9-94d4149de1e2&issueParent=99120&selectedIssue=APPS-2583) & [2582](https://uclalibrary.atlassian.net/jira/software/c/projects/APPS/boards/12?assignee=712020%3Af4539cc0-90d3-4445-9de9-94d4149de1e2&issueParent=99120&selectedIssue=APPS-2582)

https://deploy-preview-494--ucla-library-storybook-vue3x.netlify.app/?path=/docs/configure-your-project--docs

**Component Created:**. 
+ `BlockStaffSubjectLibrarian.vue`
+ `SectionStaffSubjectLibrarian.vue`

**Stories:**. 
+ `~/stories/BlockStaffSubjectLibrarian.stories.js`
+ `~/stories/SectionStaffSubjectLibrarian.stories.js`

**Spec:** ~/stories/{filename}.spec.js
+ `~/stories/SectionStaffSubjectLibrarian.spec.js`

**Notes:**

1. Refactored StaffName in the component so it does not have to be done on the page now.
2. Fixed `BlockStaffSubjectLibrarian.vue` & `BlockStaffSubjectLibrarian.stories.js` because multiple fields were misising.
  + Job Title
  + Department
  + Locations

Before
Screenshot 2024-04-08 at 3 51 41 PM
Now
Screenshot 2024-04-08 at 3 52 32 PM

--------------- ---------------

**Checklist:**

-   [ ] I checked that it is working locally in the dev server
-   [ ] I checked that it is working locally in the storybook
-   [ ] I checked that it is working locally in the 
library-website-nuxt dev server
-   [ ] I added a screenshot of it working
-   [ ] UX has reviewed and approved this
-   [ ] I assigned this PR to someone on the dev team to review
-   [ ] I used a conventional commit message
-   [ ] I assigned myself to this PR

### CODE REVIEW

File: src/lib-components/SectionStaffSubjectLibrarian.vue [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+        </th>
+      </tr>
+    </thead>
+
+    <tbody>
+      <BlockStaffSubjectLibrarian
+        v-for="(item, index) in items"
+        :key="`${index}-${item.subjectArea}`"
+        :subject-area="item.subjectArea"
+        :staff-name="item.staffName"
```
Comment (farosFreed) [inline src/lib-components/SectionStaffSubjectLibrarian.vue:45]: I think something is wrong here or somewhere with the name in SectionStaffSubjectionLibrarian. It shows okay now in the block, but in the section view it only shows the alternative name, regular first and last name are missing
Screenshot 2024-04-09 at 12 27 55 PM

**Review by farosFreed — CHANGES_REQUESTED** [review]
looks good overall, requested 1 change to make sure the name is displaying in SectionStaffSubjectLibrarian

File: src/lib-components/SectionStaffSubjectLibrarian.vue (hunk 1 above)
Comment (jendiamond) [reply src/lib-components/SectionStaffSubjectLibrarian.vue]: Refactored this.

**Review by farosFreed — APPROVED** [review]
looks great, works great 🪩🥳

**Pull request merged**